.github/scripts/e2e_tests_metrics_service.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -e
+
+LOG_FILE=".github/e2e_tests/artifacts/logs/e2e_tests/metrics_service.log"
+mkdir -p "$(dirname "$LOG_FILE")"
+
+NAMESPACE="scality-object-storage"
+SERVICE="scality-cosi-metrics"
+LOCAL_PORT=8080
+TARGET_PORT=8080
+
+# Error handling function
+error_handler() {
+  echo "An error occurred during bucket creation tests. Check the log file for details." | tee -a "$LOG_FILE"
+  echo "Failed command: $BASH_COMMAND" | tee -a "$LOG_FILE"
+  exit 1
+}
+
+# Trap errors and call the error handler
+trap 'error_handler' ERR
+
+# Log command execution to the log file for debugging
+log_and_run() {
+  "$@" 2>&1 | tee -a "$LOG_FILE"
+}
+
+log_and_run kubectl describe svc scality-cosi-driver-metrics -n scality-object-storage
+
+log_and_run kubectl port-forward svc/scality-cosi-driver-metrics -n scality-object-storage 8080:8080 &
+PORT_FORWARD_PID=$!
+
+if ps -p $PORT_FORWARD_PID > /dev/null; then
+    log_and_run echo "Port-forwarding established. Querying metrics..."
+
+    log_and_run curl -f http://localhost:$LOCAL_PORT/metrics > metrics_output.txt
+
+    if [ $? -eq 0 ]; then
```

Comment:
I think you might lose the curl return code with the pipe `| tee`
https://unix.stackexchange.com/a/14276